### ISSUE TITLE
Projet: couleurs

### ISSUE THREAD

**Issue by idris-maps** [emilie-imhof/visualisation-de-donnees#8]
Tu te compliques la vie en ajoutant du css pour le récuperer dans ton javascript.

Tu as deux possibilités

1. Uniquement javascript

Une fonction qui prends une civilisation et retourne une couleur:

```js
const couleur_region = ({ region }) => {
    switch(region) {
      case `Amériques précoloniales`:return '#EA3546'
      case `Asie de l'Est`: return  '#F86624'
      case `Asie du Sud`: return  '#F9C80E'
      case `La steppe`: return  '#BEDB39'
      case `Afrique subsaharienne`: return  '#43BCCD'
      case `Europe (et ses ramifications coloniales)`: return  '#2F8A99'
      case `Moyen-Orient`: return '#2F8A99'
      default: return '#e7e7e7'
    }
  }
```

après dans tes bâtons

```js
.attr('fill', couleur_region)
```

2. Avec CSS

Il te faut donner une classe aux bâtons en fonction de la région

```js
.attr('class', d => d.region)
```

Et ajouter ces classes avec du CSS

```css
.Moyen-Orient{ fill: #2F8A99 }
```

Le problème ici est le format de tes noms de régions. Il ne sont pas valables comme classe CSS. C'est pour ça que j'ai pris `.Moyen-Orient`. `.Asie de l'Est` par exemple ne marchera pas.

**Comment by emilie-imhof** [comment]
Oui, je vois, j'ai suivi tes conseils pour la première solution. Par contre j'avais l'impression que je passai par le switch avant et plus maintenant, car tout prenait pas défaut la couleur grise qui était dans `return ` et maintenant c'est tout en noir, donc ça veut dire que j'ai du attribuer le region_couleur au mauvais ` .style('fill')`
